### ISSUE TITLE
Java web start still referred to by GF 7.0.6 server running on Java 17

### ISSUE THREAD

**Issue by Blavo** [eclipse-ee4j/glassfish#24524]
When deploying an app client to a GF 7.0.6 server running on Java 17 the server log has references to Java web start (which was removed from Java at Java 11)

## Environment Details
Glassfish 7.0.6
java version "17.0.7" 2023-04-18 LTS
Java(TM) SE Runtime Environment (build 17.0.7+8-LTS-224)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.7+8-LTS-224, mixed mode, sharing)
Mac OS 12.4 (Monterey)
Database not relevant

----------

## Problem Description
Java Web Start was removed from (Oracle) Java at Java 11 but the log of a GF 7.0.6 server running on Java 17 still makes reference to JWS when an app client is deployed to the server.

I don't know if this is significant or not but it is confusing.

The problem is reliably reproducible.

A server log is attached.

## Steps to reproduce
Deploy a convenient application client to a convenient 7.0.6 server.

Alternatively use the steps from #<TBA>

## Impact of Issue
No impact except a bit of confusion.

[Server log.txt](https://github.com/eclipse-ee4j/glassfish/files/12206361/Server.log.txt)

**Comment by avpinchuk** [comment]
Hi,

I think this functionality could be used e.g. with OpenWebStart, which under active development now, as far I know

**Comment by Blavo** [comment]
G’day,

In the server log attached to the issue we see:

‘Java web start services started for the app client … “
and
‘Java web start services stopped for the app client …”

I suspect the issuing of these two lines is a hang over from incomplete removal of the Java web start support from glassfish and that in fact the server now doesn’t do anything with Java web services between the messages.

However, if the server is doing something useful between the two messages then there are two cases:

- that useful stuff is actually Java web start-related and the message is fine (but why are we doing something with a removed feature ?)
- that useful stuff isn’t Java web start related in which case the two messages should be amended to say what is actually being started and stopped, just to remove the confusion

**Comment by avpinchuk** [comment]
@arjantijms, @dmatej should we remove JWS support or we will try to test/update it to work with OpenWebStart?

**Comment by avpinchuk** [comment]
@Blavo, no, not only two lines in the server log, also we have a lot of code around it )

As I know, Java Web Start never been standard Java feature. It's always been proprietary extension.

**Comment by Blavo** [comment]
G’day,

According to an article at https://devm.io/java/java-web-start-openwebstart-160913#:~:text=Hendrik%20Ebbers%3A%20Unfortunately%2C%20Oracle%20not,not%20available%20as%20open%20source <https://devm.io/java/java-web-start-openwebstart-160913#:~:text=Hendrik Ebbers: Unfortunately, Oracle not,not available as open source> JWS was a standard part of Oracle Java through Java 10 as an implementation of JSR-56. According to the same article JWS was never a part of OpenJDK. It was the combination of these two facts that was the inspiration for OpenWebStart according to its instigator.

**Comment by Blavo** [comment]
G’day,

I am a bit surprised you can still support JWS at GF7 (which is pretty much why I raised the issue (which I expected to be classified as trivial)). My understanding is that JWS was removed from Oracle Java at Java 11. GF 7 supports Java 11 and 17 so is presumably built on Java 17 or later so the underlying support in the JDK shouldn’t be there. JWS never made it to OpenJDK. So I am curious which JDK GF17 is built on that can actually support JWS.

**Comment by dmatej** [comment]
> @arjantijms, @dmatej should we remove JWS support or we will try to test/update it to work with OpenWebStart?

I thought it is dead definitively, but it seems it is not: https://github.com/karakun/OpenWebStart/releases

If it would be on my decision, I would remove it from GlassFish completely and if it would be useful, then we can reintroduce some OpenWebStart integration again.

Maybe we should ask the community how many projects still use this feature in GlassFish6+ (and if it still works).
